enhancements/cluster-logging/gcp-external-account-auth.md

```diff
@@ -0,0 +1,201 @@
+---
+title: gcp-external-account-auth
+authors:
+  - "@cahartma"
+reviewers:
+  - "@alanconway"
+  - "@jcantrill"
+  - "@Clee2691"
+  - "@JoaoBraveCoding"
+  - "@xperimental"
+approvers:
+  - "@jcantrill"
+  - "@alanconway"
+api-approvers: 
+  - "@jcantrill"
+creation-date: 2025-02-27
+last-updated: 2024-02-27
+tracking-link:
+  - "https://issues.redhat.com/browse/LOG-3577"
+see-also:
+  - "https://issues.redhat.com/browse/LOG-3275"
+  - "https://issues.redhat.com/browse/LOG-6762"
+replaces: []
+superseded-by: []
+---
+
+# GCP Workload Identity Federation (WIF) Support in OpenShift Logging
+
+This enhancement enables secure, short-lived authentication for GCP services in OpenShift Logging using Workload Identity Federation
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement adds support for Workload Identity Federation (WIF) to the existing GCP 
+authentication mechanism in Vector. With this update, Vector will be able to authenticate 
+using external_account credentials when running in an OpenShift environment, enabling OIDC-based 
+authentication with short-lived access tokens instead of static credentials.
+
+## Motivation
+
+Current GCP authentication in Vector relies on service account credentials, which require 
+static key management. WIF eliminates the need for long-lived credentials, enabling more secure, 
+short-lived authentication via OpenShift service account tokens.
+
+### User Stories
+
+* As an OpenShift administrator, I want Vector to authenticate with GCP using Workload Identity Federation, 
+removing the need for static service account keys. 
+* As an engineer, I want Vector to detect the correct authentication method based on the credentials file 
+format, utilizing the correct credentials chain form the sdk, and ensuring compatibility with 
+both service accounts and external accounts (WIF).
+
+
+### Goals
+
+* Modify the GCP module in vector to support external_account credentials. 
+* Introduce new structs (ExternalCredentials, CredentialsType) to handle the updated credential format. 
+* Serialize the configuration file into usable data 
+* Implement the full OIDC-based authentication flow. 
+* Ensure compatibility with existing service account authentication
+
+
+### Non-Goals
+
+* Modifying the OpenShift gcp output authentication model. 
```

Comment:
What does this mean?  What is the "OpenShift GCP Output" auth model?

Reply:
We don't need to change any logic in CLO or in our CLF API

Reply:
Please change the verbiage here to be more specific to ClusterLogForwarder since it reads like there is some OpenShift cluster construct for GCP that is usable by our operator.

Reply:
An sts-enabled (WIF) cluster configured on GCP, is exactly an "OpenShift cluster construct for GCP that is usable by our operator".    The operator's only job at that point is to project a service account token.   The rest is up to OCP and the collector.        The only issue we have is our collector can't authenticate as it should.

Reply:
I would maybe say "Modifying the OpenShift Logging GCP output authentication Model".